changeset:   565888:91903b2fd4ab
tag:         tip
parent:      565886:1a895ef04f9f
user:        stransky <stransky@redhat.com>
date:        Thu Jan 21 12:27:43 2021 +0100
files:       widget/gtk/nsWindow.cpp
description:
Bug 1687931 [Wayland] Don't cache monitor scale factor, r?jhorak

Differential Revision: https://phabricator.services.mozilla.com/D102590


diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -7714,10 +7714,10 @@ gint nsWindow::GdkScaleFactor() {
       (gint(*)(GdkWindow*))dlsym(RTLD_DEFAULT, "gdk_window_get_scale_factor");
   if (sGdkWindowGetScaleFactorPtr && scaledGdkWindow) {
     mWindowScaleFactor = (*sGdkWindowGetScaleFactorPtr)(scaledGdkWindow);
+    mWindowScaleFactorChanged = false;
   } else {
     mWindowScaleFactor = ScreenHelperGTK::GetGTKMonitorScaleFactor();
   }
-  mWindowScaleFactorChanged = false;
 
   return mWindowScaleFactor;
 }

